compaction/compaction_manager.hh

```diff
@@ -48,9 +48,9 @@ public:
     };
     using scheduling_group = backlog_controller::scheduling_group;
     struct config {
-        scheduling_group compaction_sched_group;
-        scheduling_group maintenance_sched_group;
-        size_t available_memory;
+        scheduling_group compaction_sched_group = scheduling_group{default_scheduling_group(), default_priority_class()};
```

Comment:
Let's better patch `backlog_controller::scheduling_group` to have default-initialized parts? One of them is a reference, but it can be safely converted into a value, the `io_priority_class` is just a handle (with the class index) that can be copied around pretty much everywhere without any side effects.